scripts/automation/release/version_patcher.py

```diff
@@ -0,0 +1,77 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+#pylint: disable=line-too-long
+
+import os
+import sys
+import fileinput
+
+class VersionPatcher(object):
```

Comment:
I tend to agree with the approach in this http://davedash.com/2013/03/05/automatic-versioning-of-python-packages-with-jenkins/.

The setup.py can have a fallback for users but in build system, we should be able to avoid editing a source file.